### PR TITLE
DnD on storage

### LINKED ISSUE #2071: DnD updates in storage
These implementations need updates for storage ui

- Files and folders can be dragged into folder rows to move them
- Files and folders can be dragged into breadcrumbs to move them
- Check proper previews are created while dragging
- Renaming should prevent drags and rename textfield should allow text selections (Check FF)

### PR DESCRIPTION
**Pull request by tanmoyAtb**
closes #2071 

---
Submission checklist: 

   

#### Layout
- [x] Change looks good in the desktop web ui

#### Theme
- [x] Components / elements inspected in light mode

### CODE REVIEW

**Comment by asnaith** [comment]
Hey Tanmoy, I noticed a few issues in this branch after these changes: 

- The breadcrumb positioning/layout breaks once you start to nest folders. See video / screenshot below: 

https://user-images.githubusercontent.com/3982665/162825402-4e61a1cb-dcab-4a6c-9c22-b218f17df2a2.mov

Screen Shot 2022-04-11 at 1 25 23 PM

- The file browser is only allowing drag and drop on folders if one of them contains a file. If you try and drag and drop an empty folder into another empty folder then it doesn't work (this works on files). 

https://user-images.githubusercontent.com/3982665/162827173-9e318444-c43c-40e4-a8e9-ef1f75ab990e.mov

- I'm not seeing any DND behaviour on the mobile layout

**Comment by tanmoyAtb** [comment]
> Hey Tanmoy, I noticed a few issues in this branch after these changes:
> 
> * The breadcrumb positioning/layout breaks once you start to nest folders. See video / screenshot below:
> 
>  breadcrumb.issues.mov 
> Screen Shot 2022-04-11 at 1 25 23 PM
> 
> * The file browser is only allowing drag and drop on folders if one of them contains a file. If you try and drag and drop an empty folder into another empty folder then it doesn't work (this works on files).
> 
>  empty.folder.issue.mov 
> * I'm not seeing any DND behaviour on the mobile layout

Hey @asnaith, thanks for the check. Interesting stuff,
Can you give me the browser and screen size at which you are testing the breadcrumb breaking ?

**Comment by asnaith** [comment]
> Hey @asnaith, thanks for the check. Interesting stuff, Can you give me the browser and screen size at which you are testing the breadcrumb breaking ?

@tanmoyAtb This was at 1280 x 720 resolution on Chrome / Firefox. 

The easiest way to reproduce this is to create a folder, double click on it, add a new folder inside and repeat. After a couple of times doing that you will see that things start to go awry with the breadcrumb.

**Comment by tanmoyAtb** [comment]
All right, we had a css system missing on storage. It's all updated.
We don't really have DnD for mobile layouts on files either. So we're not going to have it here I think.

**Comment by asnaith** [comment]
Hey @tanmoyAtb this is looking great after the css fix. The only DND issue remaining is the one I mentioned before with blank folders. 

> The file browser is only allowing drag and drop on folders if one of them contains a file. If you try and drag and drop an empty folder into another empty folder then it doesn't work (this works on files).

I'm assuming we'd want to fix it (because it's allowed on Files) but I'm not sure. What do you think?

**Comment by tanmoyAtb** [comment]
> #2071

Hey @asnaith I checked on the empty folder moving functionality. It seems to be working fine on my end. 

https://user-images.githubusercontent.com/10713353/163410722-146c0b12-758f-4a6f-9110-49c7062b5948.mov

**Comment by tanmoyAtb** [comment]
> > #2071
> 
> Hey @asnaith I checked on the empty folder moving functionality. It seems to be working fine on my end.
> 
>  Screen.Recording.2022-04-14.at.8.20.56.PM.mov

I just realized the move doesn't work with IPFS file system. Looking into it.

**Comment by asnaith** [comment]
> > > #2071
> > 
> > 
> > Hey @asnaith I checked on the empty folder moving functionality. It seems to be working fine on my end.
> > Screen.Recording.2022-04-14.at.8.20.56.PM.mov
> 
> I just realized the move doesn't work with IPFS file system. Looking into it.

@tanmoyAtb Oh! Good catch. I hadn't yet realized it was specifically related to bucket type

**Comment by tanmoyAtb** [comment]
> > > > #2071
> > > 
> > > 
> > > Hey @asnaith I checked on the empty folder moving functionality. It seems to be working fine on my end.
> > > Screen.Recording.2022-04-14.at.8.20.56.PM.mov
> > 
> > 
> > I just realized the move doesn't work with IPFS file system. Looking into it.
> 
> @tanmoyAtb Oh! Good catch. I hadn't yet realized it was specifically related to bucket type

All right. It should be good to go now.

**Comment by asnaith** [comment]
> All right. It should be good to go now.

👍 Working great for both bucket types now

**Review by asnaith — APPROVED** [review]

File: packages/storage-ui/src/Components/Modules/FilesList/DragPreviewLayer.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            let Icon
+            if (previewItem.isFolder) {
+              Icon = FolderFilledSvg
+            } else if (previewItem.content_type.includes("image")) {
+              Icon = FileImageSvg
+            } else if (previewItem.content_type.includes("pdf")) {
+              Icon = FilePdfSvg
+            } else {
+              Icon = FileTextSvg
+            }
```
Comment (FSM1) [inline packages/storage-ui/src/Components/Modules/FilesList/DragPreviewLayer.tsx:175]: Can you not re-use the existing `getIcon` logic here instead?

**Review by FSM1 — APPROVED** [review]
Working really well. I left one quick suggestion, that will make for neater code, but besides that this is good to go.

File: packages/storage-ui/src/Components/Modules/FilesList/DragPreviewLayer.tsx (hunk 1 above)
Comment (tanmoyAtb) [reply packages/storage-ui/src/Components/Modules/FilesList/DragPreviewLayer.tsx:175]: all updated !

**Pull request merged**